### DISCUSSION TITLE
Can Fable be used to convert JS to F#?

### DISCUSSION THREAD

**Discussion by sumeetdas** [fable-compiler/Fable#3243] [Ideas]
I remember a tweet saying that Fable can be used to convert from any language to any language (can't find the tweet right now). So, can it be used to convert JS to F#? Has something like this done before?

**Comment by MangelMaxime** [comment]
No, I don't think fable can convert JS to F#.

It was probably saying it can convert F# to any language (as long as the language it implemented). That's why Fable can do F# to JS, F# to Rust, F# to Python, etc.

**Reply by alfonsogarciacaro** [reply]
Actually, Fable **could be** used to translate JS to F#, but not sure Fable would be very helpful in that case, because you still need to write the JS parser, transformers and then the F# printer. So still a lot of work.

When I wrote the tweet, I was mainly thinking in the hypothetical case someone would create their own F#-like language and then they could use Fable to translate it to JS, Python or Rust. But please note this was just speculative thinking and I'm not aware of anybody that is considering this idea at the moment.

**Reply by MangelMaxime** [reply]
> Actually, Fable could be used to translate JS to F#, but not sure Fable would be very helpful in that case, because you still need to write the JS parser, transformers and then the F# printer. So still a lot of work.

Well, yes I though about that but it is true for any language 😉
